UefiCpuPkg/Include/Library/SmmCpuPlatformHookLib.h

```diff
@@ -115,4 +115,22 @@ SmmCpuPlatformHookBeforeMmiHandler (
   VOID
   );
 
+/**
+  Determines whether all CPUs sync is needed.
+
+  This function determine if all CPUs needed to be synced unconditional during the 1st APs sync.
+
+  By default, it returns FALSE which indicates not all CPUs needed to be synced. In case of all CPUs
+  needed to be synced, platform specific implementation can return TRUE.
+
+  @retval TRUE   All CPUs sync is needed.
+  @retval FALSE  Not all CPUs sync is needed.
+
+**/
+BOOLEAN
+EFIAPI
+IsCpuSyncAlwaysNeeded (

```

Comment:
This function determines whether CPU Synchronization should be executed unconditionally when a SMI occurs. 

If the function returns true, it indicates that there is no need to check the system configuration and status, and CPU Synchronization should be executed unconditionally.

If the function returns false, it indicates that CPU Synchronization is not executed unconditionally, and the decision to synchronize should be based on the system configuration and status.

@retval TRUE   CPU Synchronization is executed unconditionally.
@retval FALSE  CPU Synchronization is not executed unconditionally.

Reply:
The commit message log:

This patch adds the IsCpuSyncAlwaysNeeded interface to the SmmCpuPlatformHookLib. This interface will determine whether the first CPU Synchronization should be executed unconditionally when an SMI occurs.
If the function returns true, it indicates that there is no need to check the system configuration and status, and the first CPU Synchronization should be executed unconditionally.
If the function returns false, it indicates that the first CPU Synchronization is not executed unconditionally, and the decision to synchronize should be based on the system configuration and status.

Reply:
Updated